plugin/src/main/java/com/playmonumenta/scriptedquests/quests/components/actions/dialog/DialogAllInOneEntry.java

```diff
@@ -0,0 +1,89 @@
+package com.playmonumenta.scriptedquests.quests.components.actions.dialog;
+
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.playmonumenta.scriptedquests.Plugin;
+import com.playmonumenta.scriptedquests.quests.components.QuestPrerequisites;
+import com.playmonumenta.scriptedquests.utils.MessagingUtils;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextReplacementConfig;
+import net.kyori.adventure.text.event.ClickEvent;
+import net.kyori.adventure.text.event.HoverEvent;
+
+public class DialogAllInOneEntry implements DialogBase {
+
+	// It needs an initialized value
+	private Component mComponent;
+	private String mNPCName;
+
+	public DialogAllInOneEntry(String npcName, JsonElement element) throws Exception {
+
+		JsonObject object = element.getAsJsonObject();
+		if (object == null) {
+			throw new Exception("dialog value is not an object!");
+		}
+
+		mNPCName = npcName;
+
+		Set<Entry<String, JsonElement>> entries = object.entrySet();
+		for (Entry<String, JsonElement> ent : entries) {
+			String key = ent.getKey();
+
+			JsonElement value = object.get(key);
+			if (value == null) {
+				throw new Exception("clickable_text value for key '" + key + "' is not parseable!");
+			}
```

Comment:
value should just be `JsonElement value = ent.getValue()` - there's no reason to re look it up here, which removes this value == null check.

Actually, you don't use the variable value at all - so you can just remove this entire block.